challenges/binary-search/binary_search.py

```diff
@@ -4,16 +4,17 @@ def binary_search(sorted_array, test_value):
     in sorted_array. If test_value is not in sorted_array, return
     -1 """
     counter = 0
-    if len(sorted_array) = 0:
+    if len(sorted_array) == 0:
         return -1
     else:
-        len(sorted_array > 0 and type(test_value) == int:
+        if len(sorted_array) > 0 and type(test_value) == int():
             for item in sorted_array:
                 print(counter)
                 return counter
             else:
```

Comment:
Currently this `else` statement is being executed on the `for` loop instead of the `if` statement above. Thus, when the above condition is not fulfilled your function is returning `None`.